dandi/dandiapi.py

```diff
@@ -233,6 +233,8 @@ def request(
                                 url,
                                 result.text,
                             )
+                            if data is not None and hasattr(data, "seek"):
+                                data.seek(0)
```

Comment:
could/should we may be add a test which would trigger such a case by e.g. shimming self.session.request and always failing upon first try after reading the file?

Reply:
`responses` doesn't seem to support requests where `data` is a filehandle: https://github.com/getsentry/responses/issues/719

Reply:
ok, if no easy way to test ATM, I am ok to proceed to see it fixed and others to give it a shot. Adding `release` label as well.